src/mmapped_plv.hpp

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019-2020 libsbn project contributors.
+// libsbn is free software under the GPLv3; see LICENSE file for details.
+//
+// RAII class for partial likelihood vectors that are mmapped to disk.
+//
+// This class is to allocate a very large partial likelihood vector in virtual memory
+// and then cut it up (via Subdivide) into a vector of partial likelihood vectors.
+
+#ifndef SRC_MMAPPED_PLV_HPP_
+#define SRC_MMAPPED_PLV_HPP_
+
+#include "eigen_sugar.hpp"
+#include "mmapped_matrix.hpp"
+
+using NucleotidePLV = Eigen::Matrix<double, 4, Eigen::Dynamic, Eigen::ColMajor>;
+using NucleotidePLVRef = Eigen::Ref<NucleotidePLV>;
+using NucleotidePLVRefVector = std::vector<NucleotidePLVRef>;
+
+class MmappedNucleotidePLV {
+ public:
+  MmappedNucleotidePLV(std::string file_path, Eigen::Index total_plv_length)
+      : mmapped_matrix_(file_path, 4, total_plv_length){};
+
+  NucleotidePLVRefVector Subdivide(size_t into_count) {
+    auto entire_plv = mmapped_matrix_.Get();
+    const auto total_plv_length = entire_plv.cols();
+    Assert(total_plv_length % into_count == 0,
+           "into_count isn't a multiple of total PLV length in "
+           "MmappedNucleotidePLV::Subdivide.");
+    const size_t block_length = total_plv_length / into_count;
+    NucleotidePLVRefVector sub_plvs;
+    sub_plvs.reserve(into_count);
```

Comment:
I probably don't understand how mmap works well enough to know what's going on here but won't this attempt to allocate large amounts of memory?

I was thinking more like there would be a function that requests for certain PLVs, and we just store those and return them.